spec.md

```diff
@@ -2786,8 +2884,9 @@ module = "module" , id , ":" , [ info ] , newline , indent ,
            { port , newline } ,
            { statement , newline } ,
          dedent ;
+port_extmodule = ( "input" | "output" ) , id , ":": , type_ground | type_aggregate , [ info ] ;
```

Comment:
This type of change blurs the line between syntax and semantics while also making the EBNF grammar more confusing. I.e., you likely do not have to make this differentiation because the EBNF grammar is only expressing legal syntax. If that fails type checking / type inference rules later, that's fine.

A good litmus test here is, "Should a FIRRTL _parser_ reject this code?" I would argue no.

I would not make the distinction between `port` and `port_extmodule`.

Reply:
kk, I will add `"Reset"` into `type_ground` then as well to simplify